aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -688,19 +700,15 @@ private String selectApiName(Set<String> apiClients) throws ApiException {
     private <T> SingleItemResultOperation<T> buildSingleResponseOperation(
             @NonNull String apiName,
             @NonNull GraphQLRequest<T> graphQLRequest,
-            @NonNull ResultListener<GraphQLResponse<T>> responseListener) {
+            @NonNull ResultListener<GraphQLResponse<T>> responseListener) throws ApiException {
 
         final ClientDetails clientDetails = apiDetails.get(apiName);
         if (clientDetails == null) {
-            responseListener.onError(
-                    new ApiException(
-                            "No client information for API named " + apiName,
-                            "Check your amplify configuration to make sure there " +
-                            "is a correctly configured section for " + apiName
-                    )
+            throw new ApiException(
+                    "No client information for API named " + apiName,
+                    "Check your amplify configuration to make sure there " +
```

Comment:
amplifyconfiguration.json

Reply:
I think it's better to leave it like this since the user could be passing in a custom configuration not through the JSON file.